### DISCUSSION TITLE
关于新版本搜索功能的建议。

### DISCUSSION THREAD

**Discussion by somaicro** [tamlok/vnote#1743] [Ideas]
一直使用着2.91版本，觉得搜索高亮的功能很棒，但是新版本没有有点遗憾，但是新版本也是很棒，毋庸置疑。
1. 能否在新的版本中，加入搜索高亮功能，让点击位置列表信息后，搜索信息过后一目了然。
2. 搜索范围增加一个，当前 .md这个笔记范围。（有看到缓冲区，但是是打开的所有.md。）有时候只想一个文件里搜素，而不是整一个笔记本。
感谢您的默默付出。

**Comment by tamlok** [comment]
1. 高亮暂时没有时间弄，后面会添加；
2. 一个文件的话，是不是用页面查找`Ctrl+F`会比较方便？

**Reply by somaicro** [reply]
是的ctrl + F是一个解决的方式，老版本中就是如此。但新版本这个功能变成了独立的，即是每个md都是独立一个页面查找。，，或者在新版本的合并查询功能中，然后用查询范围作为功能分割。毕竟，两者的作用是相近的。

**Reply by tamlok** [reply]
还是有点蒙，你既要单独查找一个文件，但独立一个页面查找又有问题。不管怎么样，后面加上吧，你是需要当前缓冲区，还是当前文件？

**Reply by tamlok** [reply]
合并是不可能合并的，后面每个编辑窗口都可以弹出来作为一个独立窗口的，所以必须有独立的页面查找。

**Reply by somaicro** [reply]
可能表达不是很清晰，意思是指查找页面会一直存在直到不需要的时候才关闭，不会因为切换页面而消失。

**Reply by tamlok** [reply]
esc可以关闭？

**Reply by tamlok** [reply]
另外， 试了一下，3.0的是每个页面独立一个窗口的。

**Reply by somaicro** [reply]
嗯嗯是的，3.0每个页面都是独立得，不过，觉得2.91一样页面搜索不是独立的更方便。
因为独立的话，每一次切换界面得重新ctrl+F，重新输入查找内容，这个操作很繁琐。
话说作者大大，新版本会考虑携带版么，就是用户资料和设置，存放在程序目录里。因为程序默认存放在C盘Users目录下，周转公司和家里，用户资料和设置都不能携带。

**Reply by tamlok** [reply]
数据自己随便放哪里都行，同步好，可以再次导入到vnote中。配置文件一般没必要同步。确实要同步，可以看文档有说明怎么搞。

**Reply by somaicro** [reply]
好的  感谢。
